core/src/main/scala/kafka/network/RequestConvertToJson.scala

```diff
@@ -0,0 +1,362 @@
+      } else {
+        node.set("transactionalId", new TextNode(request.transactionalId))
+      }
+    }
+    node.set("acks", new ShortNode(request.acks))
+    node.set("timeoutMs", new IntNode(request.timeout))
+    val topics = CollectionUtils.groupPartitionDataByTopic(request.partitionRecordsOrFail())
+    val topicsArray = new ArrayNode(JsonNodeFactory.instance)
+    topics.forEach { (topicName, partitions) =>
+      val topicNode = new ObjectNode(JsonNodeFactory.instance)
+      topicNode.set("name", new TextNode(topicName))
+      val partitionsArray = new ArrayNode(JsonNodeFactory.instance)
+      partitions.forEach { (partitionIndex, partitionData) =>
+        val partitionNode = new ObjectNode(JsonNodeFactory.instance)
+        partitionNode.set("partitionIndex", new IntNode(partitionIndex))
+        if (partitionData == null) {
+          partitionNode.set("records", NullNode.instance)
+        } else {
+          if (serializeRecords)
+            partitionNode.set("records", new BinaryNode(util.Arrays.copyOf(partitionData.buffer().array(), partitionData.validBytes())))
+          else
+            partitionNode.set("records", new IntNode(partitionData.validBytes()))
+        }
+        partitionsArray.add(partitionNode)
+      }
+      topicNode.set("partitions", partitionsArray)
+      topicsArray.add(topicNode)
+    }
+    node.set("topics", topicsArray)
+    node
+  }
+
+  /**
+   * Temporary until switch to use the generated schemas.
+   */
+  def offsetsForLeaderEpochResponseNode(response: OffsetsForLeaderEpochResponse, version: Short): JsonNode = {
+    val node = new ObjectNode(JsonNodeFactory.instance)
+    if (version >= 2) {
+      node.set("throttleTimeMs", new IntNode(response.throttleTimeMs))
+    }
+    val topics = CollectionUtils.groupPartitionDataByTopic(response.responses)
+    val topicsArray = new ArrayNode(JsonNodeFactory.instance)
+    topics.forEach { (topicName, partitions) =>
+      val topicNode = new ObjectNode(JsonNodeFactory.instance)
+      topicNode.set("name", new TextNode(topicName))
+      val partitionsArray = new ArrayNode(JsonNodeFactory.instance)
+      partitions.forEach { (partitionIndex, partitionData) =>
+        val partitionNode = new ObjectNode(JsonNodeFactory.instance)
+        partitionNode.set("errorCode", new ShortNode(partitionData.error.code))
+        partitionNode.set("partitionIndex", new IntNode(partitionIndex))
+        if (version >= 1) partitionNode.set("leaderEpoch", new IntNode(partitionData.leaderEpoch))
+        partitionNode.set("endOffset", new LongNode(partitionData.endOffset))
+        partitionsArray.add(partitionNode)
+      }
+      topicNode.set("partitions", partitionsArray)
+      topicsArray.add(topicNode)
+    }
+    node.set("topics", topicsArray)
+    node
+  }
+
+  /**
+   * Temporary until switch to use the generated schemas.
+   */
+  def produceResponseNode(response: ProduceResponse, version: Short): JsonNode = {
```

Comment:
Seeing as https://github.com/apache/kafka/pull/9401 is about to merge, maybe we can wait for that before merging this and drop our manual JSON generation code for produce request/response?

Reply:
We are actually waiting on that one and this one https://github.com/apache/kafka/pull/9547.

Reply:
Great

Reply:
Yes! we're waiting on #9401 and #9547 to merge. Once they're merged, I'll pick this PR back up and will remove all the temporary case handling done for the Produce and OffsetsForLeaderEpoch request/response.